base/server/src/main/java/com/netscape/cmscore/ldapconn/LdapAnonConnFactory.java

```diff
@@ -201,18 +201,20 @@ protected void init() throws ELdapException {
         logger.debug("LdapAnonConnFactory: secure: " + mConnInfo.getSecure());
 
         // initalize minimum number of connection handles available.
-        makeMinimum(mErrorIfDown);
+        if (mMinConns > 0) {
+            makeMinimum(mErrorIfDown);
+        }
         mInited = true;
     }
 
     /**
      * make the mininum configured connections
      */
     protected void makeMinimum(boolean errorIfDown) throws ELdapException {
-
+        int realMin = Math.max(mMinConns, 1);
         try {
-            if (mNumConns < mMinConns && mTotal < mMaxConns) {
-                int increment = Math.min(mMinConns - mNumConns, mMaxConns - mTotal);
+            if (mNumConns < realMin && mTotal < mMaxConns) {
+                int increment = Math.min(realMin - mNumConns, mMaxConns - mTotal);
```

Comment:
Hmm.. I thought if `mMinConns == 0` the original code would not create a new connection anyway, is that not the case?

Reply:
Ah, I see that you're using `makeMinimum()` to create at least 1 connection.

Reply:
I use it when the connection is requested so I have not change the following code but there are no connection waiting